src/Xrefcheck/Core.hs

```diff
@@ -17,6 +17,7 @@ import Data.Default (Default (..))
 import qualified Data.List as L
```

Comment:
I tried modifying my server to _always_ wait 2 seconds, and then _always_ respond with a 429 and `Retry-After: 2`.

I'm now seeing this behaviour:

1. The progress bar's size keeps increasing every time a 429 is retried.
2. When the window's width is narrow enough, xrefcheck floods stdout.

https://user-images.githubusercontent.com/718166/140304183-a6db1443-5518-4067-a28b-f375dcf52006.mp4

Reply:
> 2. When the window's width is narrow enough, xrefcheck floods stdout.

This is the old behaviour btw, probably this can't be fixed without special means.

Reply:
> 1. The progress bar's size keeps increasing every time a 429 is retried.
Oh yeah, I've encountered this problem as well. Will fix that.

Reply:
It seems the 2nd problem was already there in `master` and was not introduced here.

I checked out master, create a file with 10 links (the first takes 1s to complete, the 2nd takes 2s, the 3rd 3s, etc), and I observed a similar behaviour.

https://user-images.githubusercontent.com/718166/140309350-0b410f4e-2fde-4859-a510-650a7cbb0be8.mp4

I'll create a separate issue for this and we'll tackle it later.

Reply:
> When the window's width is narrow enough, xrefcheck floods stdout.
This is the old behaviour btw, probably this can't be fixed without special means.

Oh I didn't refresh my window, so I didn't see your reply before posting mine :sweat_smile: 

TBH this is probably not a big deal... As long as we keep our output within the normal ~80 (?) width limit, it should be good enough. Maybe this is not worth creating an issue.